wallet/src/bin/faucet_server.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2022 Espresso Systems (espressosys.com)
+// This file is part of the Configurable Asset Privacy for Ethereum (CAPE) library.
+
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # The CAPE Faucet
+//!
+
+extern crate cape_wallet;
+
+use async_std::{
+    sync::{Arc, Mutex},
+    task::{spawn, JoinHandle},
+};
+use cape_wallet::{
+    routes::{init_wallet, Wallet},
+    wallet::CapeWalletError,
+    web::default_storage_path,
+};
+use jf_cap::{
+    keys::{UserKeyPair, UserPubKey},
+    structs::AssetCode,
+};
+use rand_chacha::{rand_core::SeedableRng, ChaChaRng};
+use seahorse::loader::Loader;
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::path::PathBuf;
+use structopt::StructOpt;
+use tide::StatusCode;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "CAPE Faucet Server",
+    about = "Grants a native asset seed to a provided UserPubKey"
+)]
+pub struct FaucetOptions {
+    /// mnemonic for the faucet wallet
+    #[structopt(long = "mnemonic", env = "MNEMONIC", default_value = "")]
+    pub mnemonic: String,
+
+    /// path to the faucet wallet
+    #[structopt(long = "faucet_wallet_path", env = "FAUCET_WALLET_PATH")]
+    pub faucet_wallet_path: PathBuf,
```

Comment:
I had some issues trying to run the the binary.

I think the first issue is that `write_path` may try to write `last_wallet_path` to a directory that doesn't exist (if one did not run the wallet web server, resp. `create_dir_all` inside the wallet's `init_server` before).

If I do make sure the directory is there I get
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to open wallet: cannot open wallet that does not exist', wallet/src/bin/faucet_server.rs:143:14
```
Is there some initialization code I should run to create the faucet wallet?

Reply:
TBH: I am gonna need a more clear explainer of how the (configure/setup/run) loop is supposed to work currently to give any meaningful feedback.

Reply:
I think we may need to discuss first on what this workflow should look like. I'll add it to the discussion topic for the meeting today.